authentication/auth_method_oidc.go

```diff
@@ -65,19 +68,43 @@ func (a oidcAuth) getMSALToken(ctx context.Context, api environments.Api, _ auto
 		return &authWrapper.Authorizer{Authorizer: conf.TokenSource(ctx)}, nil
 	}
 
+	idToken := a.idToken
+
+	if a.idTokenFilePath != "" {
+		idToken, err = a.readTokenFile(a.idTokenFilePath)
+
+		if err != nil {
+			return nil, fmt.Errorf("reading token file: %v", err)
+		}
+
+		if a.idToken != "" && a.idToken != idToken {
+			return nil, fmt.Errorf("mismatch between supplied OIDC token and supplied OIDC token file")
```

Comment:
@katbyte have added an error when the token doesn't match what is read from the token file based on your recommendation in https://github.com/hashicorp/go-azure-helpers/pull/131#discussion_r980315532. Let me know if you think this text should be adjusted at all!

Reply:
would it be a good idea to output the tokens?